examples/lua/bashreadline.lua

```diff
@@ -0,0 +1,30 @@
+--[[
+Copyright 2016 GitHub, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]
+local ffi = require("ffi")
+
+return function(BPF)
+  local b = BPF:new{src_file="bashreadline.c", debug=true}
```

Comment:
Thinking aloud here...if we do truly share the .c file implementation, then duplicating that code in the different bindings could become a bad thing. We should weigh that duplication against tool portability.

Possible solutions to a problem that only just barely exists yet: cmake prepocessing, .c implementation moving to libbcc, git precommit hook to detect out-of-sync files, etc.